src/isomorphic/classic/class/ReactClass.js

```diff
@@ -728,6 +728,11 @@ var ReactClassMixin = {
    * type signature and the only use case for this, is to avoid that.
    */
   replaceState: function(newState, callback) {
+    if (this.updater.isFiberUpdater) {
```

Comment:
Can you explain why Fiber needs to be special here and take a callback argument instead of a separate `enqueueCallback` call?

If we're going down that route, we should just do the same for the stack reconciler instead.